### PR TITLE
Refactor(eos_designs,eos_cli_config_gen): Default validation_mode to error and remove conversion_mode

### LINKED ISSUE #4284: Remove support for automatic conversion from dict-of-dicts to list-of-dicts
Remove support for automatic conversion from dict-of-dicts to list-of-dicts in both `eos_designs` and `eos_cli_config_gen`. This should be split into multiple PRs as needed.

`eos_designs`:
- [x] Remove `convert_types: [dict]` from schema
- [x] Remove `convert_dicts()` from Python code.
- [x] Clean up deprecated tests if they are only testing the old dict models.

`eos_cli_config_gen`:
- [x] Remove `convert_types: [dict]` from schema
- [x] Remove `| convert_dicts` from Jinja templates.
- [x] Clean up deprecated tests if they are only testing the old dict models.

General
- [x] Identify schemas with `convert_types: [list]` since these may also rely on `convert_dicts` code.
- Remove `avd_data_conversion_mode` from
  - [x] schemas
  - [x] Ansible plugins
  - [x] PyAVD
  - [x] Documentation
- [x] Update release-notes + porting-guide with this change
- [x] Clean up schema tooling to longer convert dicts and lists and no longer emit deprecation and conversion warnings

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Default validation_mode to error and remove conversion_mode

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.<role-name>`

### User Checklist

- N/A

### Repository Checklist

- [ ] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py [hunk 1]
```diff
@@ -180,7 +143,6 @@ def handle_validation_exceptions(self, exceptions: Generator, mode: str) -> int:
                 continue
 
             if isinstance(exception, AvdDeprecationWarning):
-                # Deprecation warnings are not subject to "conversion_mode".
                 # Instead we display using Ansible's deprecation notices.
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py]: I think this comment should be modified, it seems incomplete.

File: ansible_collections/arista/avd/docs/contribution/eos_designs_internal_notes.md [hunk 2]
```diff
@@ -43,8 +43,7 @@ The module is used in `arista.avd.eos_designs` to set facts for devices, which a
 
 ```yaml
   template_output: <true | false | default -> false>
-  conversion_mode: <"warning" | "info" | "debug" | "quiet" | "disabled" | default -> "debug">
-  validation_mode: <"error" | "warning" | "info" | "debug" | "disabled" | default -> "warning">
+  validation_mode: <"error" | "warning" default -> "error">
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/docs/contribution/eos_designs_internal_notes.md]:
```suggestion
  validation_mode: <"error" | "warning" | default -> "error">
```

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py]: Updated

**Review by MaheshGSLAB — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM, tested with inventory.

**Pull request merged**